node/node.go

```diff
@@ -55,84 +55,67 @@ const (
+	NOTIFY                 = "notify"                 // Used to notify a node about a new predecessor.
+	PUT                    = "put"                    // Used to insert a DNS query.
+	GET                    = "get"                    // Used to retrieve a DNS record.
+	GETSOME                = "get_some"               // Used to shift entries.
+	EMPTY                  = "empty"                  // Placeholder or undefined message type or errenous communications.
+	REPLICATE              = "replicate"              // Used to replicate data.
 )
 
 /*
 The default method called by all RPCs. This method receives different
 types of requests, and calls the appropriate functions.
 */
 func (node *Node) HandleIncomingMessage(msg *message.RequestMessage, reply *message.ResponseMessage) error {
-	if node.Logging {
-		systemcommsin.Println("Message of type", msg.Type, "received.")
-	}
+	log.Info().Msgf("Message of type %s received.", msg.Type)
 	switch msg.Type {
 	case PING:
-		if node.Logging {
-			systemcommsin.Println("Received ping message")
-		}
+		log.Info().Msg("Received PING message")
 		reply.Type = ACK
 	case FIND_SUCCESSOR:
-		if node.Logging {
-			systemcommsin.Println("Received a message to find successor of", msg.TargetId)
-		}
+		log.Info().Msgf("Received a message to FIND SUCCESSOR of %d", msg.TargetId)
 		pointer, _ := node.FindSuccessor(msg.TargetId, msg.HopCount)
 		reply.Type = ACK
 		reply.Nodeid = pointer.Nodeid
 		reply.IP = pointer.IP
 	case NOTIFY:
-		if node.Logging {
-			systemcommsin.Println("Received a message to notify me about a new predecessor", msg.TargetId)
-		}
+		log.Info().Msgf("Received a message to NOTIFY me about a new predecessor %d", msg.TargetId)
 		status := node.Notify(Pointer{Nodeid: msg.TargetId, IP: msg.IP})
 		if status {
 			reply.Type = ACK
 		}
 	case GET_PREDECESSOR:
-		if node.Logging {
-			systemcommsin.Println("Received a message to get predecessor")
-		}
+		log.Info().Msg("Received a message to GET PREDECESSOR")
 		reply.Nodeid = node.Predecessor.Nodeid
 		reply.IP = node.Predecessor.IP
 	case GET:
-		if node.Logging {
-			systemcommsin.Println("Received a message to Get DNS record")
-		}
+		log.Info().Msg("Received a message to GET DNS record")
 		reply.QueryResponse = node.GetQuery(msg.TargetId)
 	case GETSOME:
-		systemcommsin.Println("Received a message to Get some DNS records")
+		log.Info().Msg("Received a message to GET SOME DNS records")
 		reply.Payload = node.GetShiftRecords(msg.TargetId)
 	case PUT:
-		if node.Logging {
-			systemcommsin.Println("Recieved a message to insert a query")
-		}
+		log.Info().Msg("Received a message to INSERT a query")
 		status := node.PutQuery(msg.TargetId, msg.Payload)
 		if status {
 			reply.Type = ACK
 		}
 	case REPLICATE:
-		if node.Logging {
-			systemcommsin.Println("Recieved a message to replicate data")
-		}
+		log.Info().Msg("Received a message to REPLICATE data")
```

Comment:
We can move these to DEBUG level also I think. Since most of these messages are being printed in CallRPC anyway --> which has a standard format.